src/transaction_base.js

```diff
@@ -0,0 +1,202 @@
+import xdr from './generated/stellar-xdr_generated';
+import { hash } from './hashing';
+import { Keypair } from './keypair';
+
+/**
+ * @ignore
+ */
+export class TransactionBase {
+  constructor(tx, signatures, fee, networkPassphrase) {
+    if (typeof networkPassphrase !== 'string') {
```

Comment:
we were suppose to make this change on the previous major bump but we forgot, not this is a good time.